### PR TITLE
gandi improvements

### PR DESCRIPTION
**Pull request by TinfoilSubmarine**

### CODE REVIEW

**Comment by LenardHess** [comment]
When using `wantipv4/wantipv6`, the provider implementation has to then set `status-ipv4/status-ipv6` instead of `status`, `ipv4/ipv6` instead of `ip` and so on.

I admit the rules for this aren't clearly written down yet, we went over some details in this pull request (while in a fork, so other repo): https://github.com/rrthomas/ddclient/pull/10

**Comment by TinfoilSubmarine** [comment]
Is this better? I also added skipping the update if the record is already set to the same IP.

**Comment by TinfoilSubmarine** [comment]
FYI, I'm done making revisions on this.

**Comment by LenardHess** [comment]
Thanks for the update (again :wink:)

We will leave this pull request open until v3.11.0 has been released (see https://github.com/ddclient/ddclient/issues/552#issuecomment-1675925270 for details). It will get pulled into v3.11.1

**Comment by LenardHess** [comment]
Same as #557 - please rebase onto master for the merge.

File: ddclient.in [hunk 1]
```diff
@@ -7427,20 +7427,20 @@ Available configuration variables:
...
 
   ## Multiple host update.
-  protocol=gandi,                           \\
-  zone=example.com,                         \\
-  password=my-gandi-api-key,                \\
-  ttl=1h                                    \\
+  protocol=gandi
+  zone=example.com
+  password=my-gandi-api-key
+  ttl=360 # optional
```
[the remark this answers is not in the document]
Comment (TinfoilSubmarine) [reply ddclient.in]: thanks, updated

File: ddclient.in [hunk 2]
```diff
@@ -7357,68 +7357,87 @@ EoEXAMPLE
...
+            if (!defined($response)) {
+                $config{$h}{"status-$ipv"} = "bad";
 
-            failed("%s -- Unexpected service response.", $h);
-            next;
-        }
+                failed("%s -- Unexpected service response.", $h);
+                next;
+            }
+            if($response->{'rrset_values'}->[0] eq $ip) {
```
Comment (LenardHess) [inline ddclient.in:7402]: This should also check the TTL

File: ddclient.in [hunk 3]
```diff
@@ -7357,68 +7357,87 @@ EoEXAMPLE
...
+                success("updating %s: skipped: address was already set to %s.", $h, $ip);
+                next;
```
Comment (LenardHess) [inline ddclient.in:7404]: This has to set 'status-ipv*', 'ipv*' and 'mtime' to properly let ddclient cache/avoid redundant update checks.

**Review by LenardHess — CHANGES_REQUESTED** [review]
There is a few things i still noticed - Apologies for not noticing them the first time around.
The "Nitpick" stuff isn't urgent, feel free to skip those suggestions.

**Review by LenardHess — APPROVED** [review]
Looks good :+1:

**Pull request merged**